core/core/src/sf_core.rs

```diff
@@ -203,19 +211,23 @@ impl OneClientCore {
         //     tracing::error!("Input validation error: {}", err);
         // }
 
-        // TODO: Validate Parameters
+        // Validate parameters values against json schema
+        self.parameters_validator
+            .validate(&perform_input.map_parameters)
+            .map_err(|err| {
+                PerformException::from_json_schema_validation_error(
+                    err,
+                    Some("Parameters".to_string().as_ref()),
+                )
+            })?;
+
         let mut map_parameters = match perform_input.map_parameters {
```

Comment:
I wanted to move it down to line 257 close to provier parameters, but on line 247 is self moved :/ and there is used `self.host_value_to_map_value`. 

I remember we talked why `host_value_to_map_value` is struct method, just don't remember outcome. Anyways, I guess it can stay here.

Reply:
Technically we could untangle it because `host_value_to_map_value` will only need to borrow the fields which will store host-to-map mapping state (for streams and such) while line 247 borrows `self` because of cache.

Alternativelt we could make two versions of `host_value_to_map_value` - one that would take state and handle streams and one that wouldn't and would ignore streams (which can't appear in provider parameters anyway).